pingcap/tidb-operator
# Fix the issue that backup files may leak after cleanup

### What problem does this PR solve?


### What is changed and how does it work?


1. Add a field `retryCount`, and retry to clean backup files when one cleanup fails
2. Try to clean all files in one cleanup instead of returning when met any errors.
3. Add a field `backoffEnabled` to enable backoff when a deletion API fails. It may be useful when the deletion API is rate limited.

### Code changes


### Tests



1. Create a Backup and add 40w files to the bucket.
2. Delete Backup.
3. Check all files are deleted.




### Side effects


### Related changes


### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Fix the issue that backup files leak after cleanup
```
